### ISSUE TITLE
Package scope

### ISSUE THREAD

**Issue by cdvv7788** [swappsco/django-plans#13]
I think that one of the main issues with the package right now is scope. Currently i feel it does too much, when it should try to handle only one thing and handle it well.
It is extensible, yes, but still takes a lot of time to install, setup and integrate, specially the payments.

I think that to move forward with the project we need a way to improve that. Either reduce the scope or improve the initial setup flow.

**Comment by cdvv7788** [comment]
If @rdlabspl and @shnela are using this package, we would like to hear your opinion on this.

**Comment by rdlabspl** [comment]
@shnela is part of @rdlabs, I will be using my company account.

I personally didn't have any problems with integrating your library with my project.
In fact I had more problems with implementing additional features, and that's why I've created other issues in this project.
Setting payments up isn't trivial, but I think that Fake payments view is the best what we can do in this project.

Which parts of this tool do you want to remove?

**Comment by cdvv7788** [comment]
I still need to give a full review to the package, but I am not sure if we should be handling invoicing for example. All the package should care about is creating the orders and keeping it's state, notifying the rest of the system about it. This way we can integrate with other invoicing solution and not forcing it on the user, as we are not creating django-invoices.
Also, it is painful to setup the package on an existing project. A flow review/cleanup should help to remove some of that pain tho.

**Comment by rdlabspl** [comment]
@cdvv7788
>  I am not sure if we should be handling invoicing for example

I agree, many users may want to implement their own invoicing system.
How do you want to manage this? Just remove the whole invoicing section and forget about it, or maybe create django-plans-invoice plugin?
What about users who are using our invoicing system? You have to remember that this package is listed in the Python Package Index.

**Comment by sebastian-code** [comment]
It's a great point. The backwards compatibility can be a great pain, more so if not done in the best way. Isn't there a django-invoice package already?

**Comment by cdvv7788** [comment]
So we can move that model to another package and provide a migration path.

**Comment by sebastian-code** [comment]
Ok, but I think that needs to be properly planed to avoid headaches. Looking out there for invoice packages I found a few:

1. [django-invoice](https://github.com/simonluijk/django-invoice)
2. [django-simple-invoice](https://github.com/mwolff44/django-simple-invoice)
3. [django-invoincing](https://github.com/PragmaticMates/django-invoicing)
4. [django-silver](https://github.com/silverapp/silver)

All of them have different levels of development. with ones being really interesting. Perhaps we should consider some kind of easy integration flow with this ones. Or perhaps not.
